logic/constants.py

```diff
@@ -8,7 +11,18 @@
 
 EVERYTHING = EIN("Everything")
 
-MAX_HINTS = 32
+NUMBER_OF_HINT_STONES = 16
+HINT_DISTROS_DIRECTORY = RANDO_ROOT_PATH / f"hints/distributions"
+
+hint_distros = [f for f in listdir(HINT_DISTROS_DIRECTORY) if f.endswith("json")]
+max_hints_per_stone = 0
+for distro_filename in hint_distros:
+    with open(HINT_DISTROS_DIRECTORY / distro_filename) as f:
+        hints_per_stone = json.load(f)["hints_per_stone"]
+        if hints_per_stone > max_hints_per_stone:
+            max_hints_per_stone = hints_per_stone
+
+MAX_HINTS = max_hints_per_stone * NUMBER_OF_HINT_STONES
```

Comment:
```suggestion
MAX_HINTS_PER_STONE = 8
MAX_HINTS = MAX_HINTS_PER_STONE * NUMBER_OF_HINT_STONES
```

That also means we can avoid reading all the distros here